controllers/chat/moveToAnon.js

```diff
@@ -218,14 +218,14 @@ const moveToAnon = async (req, res) => {
       sort: [{updated_at: -1}],
       limit: 100
     });
-    const {betterChannelMember, betterChannelMemberObject} =
+    const {betterChannelMember, betterChannelMemberObject, updatedChannel} =
       await BetterSocialCore.chat.updateBetterChannelMembers(newChannel, createdChannel, true, {
         channel_type: CHANNEL_TYPE.ANONYMOUS
       });
 
     await client.disconnectUser();
     const response = {
-      ...createdChannel,
+      ...updatedChannel,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes look good. The `updatedChannel` is now being used in the response object instead of `createdChannel`. This ensures that the most recent state of the channel is returned in the response. However, please ensure that `updatedChannel` always contains the latest state of the channel after all operations are performed on it.
```diff
-      ...createdChannel,
+      ...updatedChannel,
```
This change improves the accuracy of the data returned to the client by ensuring that the most up-to-date channel information is included in the response.


<!-- This is an auto-generated comment by OSS CodeRabbit -->